Project/Assets/ML-Agents/Examples/Crawler/Scripts/CrawlerAgent.cs

```diff
@@ -9,13 +10,47 @@
 [RequireComponent(typeof(JointDriveController))] // Required to set joint forces
 public class CrawlerAgent : Agent
 {
-    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
-    Vector3 m_WalkDir; //Direction to the target
-    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    //The type of crawler behavior we want to use.
+    //This setting will determine how the agent is set up during initialization.
+    public enum CrawlerAgentBehaviorType
+    {
+        CrawlerDynamic,
+        CrawlerDynamicVariableSpeed,
+        CrawlerStatic,
+        CrawlerStaticVariableSpeed
+    }
+
+    public CrawlerAgentBehaviorType typeOfCrawler;
```

Comment:
no change. if changed at run time nothing would change since it's called in initialize. i didn't consider your case. I assumed the agent behavior type wouldn't be changed during a training session. lets leave as-is for now and we can expand this functionality if needed. may be a good idea as we add more agent abilities

Reply:
What about changing the setter on the property so the `typeOfCrawler` cannot be changed after Initialize is called?

Reply:
no change. would require a custom editor